client/src/store/thunk/todos.ts

```diff
@@ -0,0 +1,20 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import axios, { AxiosError } from "axios";
+
+import { ITask } from "@Types/task";
+
+export const fetchTodos = createAsyncThunk(
+  "todos/fetchTodos",
+  async (token: string, { rejectWithValue }) => {
+    try {
+      const response = await axios.get<ITask[]>(
+        `${process.env.REACT_APP_API_URL}/task`,
+        { headers: { Authorization: `Bearer ${token}` } }
```

Comment:
Should be in global config for authorized requests, other way you will need to duplicate it for all requests where you need token